pokepro/src/components/InputBar/InputBar.tsx

```diff
@@ -1,24 +1,44 @@
-import React, { useState, ChangeEvent, useEffect } from "react";
-import { debounce } from "../../utils/debounce";
+import React, { useState, ChangeEvent, useEffect, useRef } from "react";
+import { useNavigate } from "react-router-dom";
 import { BsSearch } from "react-icons/bs";
 import { BiHomeHeart } from "react-icons/bi";
-import {Link} from "react-router-dom"
+import { Link } from "react-router-dom";
+import { pokemonSpeciesRoute } from "../../utils/baseUrl";
+import { debounce } from "lodash";
+import axios from "axios";
 function InputBar() {
+  const inputRef = useRef("");
+  const navigate = useNavigate();
   const [searchInput, setSearchInput] = useState("");
-  const [debouncedValue, setDebouncedValue] = useState("");
-  const handleChange = debounce((value: string) => console.log(value), 1000);
+  const [debouncedValue, setDebouncedValue] = useState([]);
+  async function search(name: string) {
+    const response = await axios.get(`${pokemonSpeciesRoute}/${name}`);
+    console.log(response);
+    if (response.status === 200) {
+      let ID = response.data.id;
+      navigate(`/pokemon/${ID}`);
+    }
+  }
+  const debouncedSearch = React.useRef(
+    debounce(async (criteria) => {
+      await search(criteria);
+    }, 1000)
+  ).current;
```

Comment:
useCallback instead and reduce the debounce time to 600 ms.

```suggestion
  const debouncedSearch = useCallback(
    debounce(search, 600)
  , []);
```